cmd/akashctl/provider.go

```diff
@@ -62,34 +86,117 @@ func providerCmd(cdc *codec.Codec) *cobra.Command {
 				return err
 			}
 
-			log := log.NewTMLogger(log.NewSyncWriter(os.Stdout))
+			lg := log.NewTMLogger(log.NewSyncWriter(os.Stdout))
 
-			session := session.New(log, aclient, pinfo)
+			ses := session.New(lg, aclient, pinfo)
 
 			bus := pubsub.NewBus()
 			defer bus.Close()
 
-			pubdone := make(chan error, 1)
+			var cclient cluster.Client
+			var service provider.Service
 
-			go func() {
-				pubdone <- events.Publish(ctx, cctx.Client, "provider-cli", bus)
-			}()
+			// todo (troian) provider selection
+			cclient = cluster.NullClient()
 
-			var cclient cluster.Client
+			if service, err = provider.NewService(ctx, ses, bus, cclient); err != nil {
+				return err
+			}
 
-			service, err := provider.NewService(ctx, session, bus, cclient)
-			if err != nil {
+			var rest *serviceRest
+			if rest, err = newServiceRest(serviceRestConfig{
+				ctx:           ctx,
+				log:           lg,
+				serviceStatus: service,
```

Comment:
would want to pass `service` into this so that it can call, say, `service.Status()`.